src/vs/editor/common/viewLayout/viewLineRenderer.ts

```diff
@@ -48,6 +52,7 @@ export class RenderLineInput {
 	public readonly renderWhitespace: RenderWhitespace;
 	public readonly renderControlCharacters: boolean;
 	public readonly fontLigatures: boolean;
+	public readonly selectionsOnLine?: Range[];
```

Comment:
Just a nitpick: I would prefer to use `Range[] | null` instead of `Range[] | undefined`. 

I personally prefer `null` to `undefined` because `undefined` can "sneak in" by accident by omitting to pass an argument, etc. while I find `null` to be more intentional since it must be spelled out in all call sites, etc. This is not a problem with this PR, but some future users might omit the argument by accident etc..